trio/_tools/gen_exports.py

```diff
@@ -24,11 +24,25 @@
 HEADER = """# ***********************************************************
 # ******* WARNING: AUTOGENERATED! ALL EDITS WILL BE LOST ******
 # *************************************************************
-# isort: skip
+# isort: skip_file
+from __future__ import annotations
+
+import sys
+from typing import TYPE_CHECKING, Callable, ContextManager
+
 from ._instrumentation import Instrument
 from ._ki import LOCALS_KEY_KI_PROTECTION_ENABLED
 from ._run import _NO_SEND, GLOBAL_RUN_CONTEXT
 
+if TYPE_CHECKING:
+    import select
+    from socket import socket
+
+    from ._traps import Abort, RaiseCancelT
+
+    from .. import _core
+    from ._unbounded_queue import UnboundedQueue
```

Comment:
In my `run` PR to avoid adding a lot of irrelevant imports to this file I just `import trio`, then changed the original types to do `trio.whatever.Class`. Perhaps we should just change the processing logic below to have a different set of imports for each file instead though.